host/CMakeLists.txt

```diff
@@ -95,16 +98,22 @@ else()
   message(FATAL_ERROR "Unknown OS. Only supported OSes are Linux and Windows")
 endif()
 
-# SGX specific files
+# SGX specific files.
 if (OE_SGX)
-  list(APPEND PLATFORM_SRC
+  set(NEW_HOSTVERIFY_SRC
+    ../common/sgx/tlsverifier.c
     ../common/sgx/qeidentity.c
     ../common/sgx/quote.c
     ../common/sgx/report.c
     ../common/sgx/revocation.c
     ../common/sgx/sgxcertextensions.c
     ../common/sgx/tcbinfo.c
-    ../common/sgx/tlsverifier.c
+    sgx/sgxquoteprovider.c
+    sgx/hostverify_report.c)
+
+  list(APPEND PLATFORM_HOSTVERIFY_SRC ${NEW_HOSTVERIFY_SRC})
```

Comment:
Agreed. I'll try to simply it. Ideally, we should simplify it even further; the platform specific folders should have their own CMakeLists.txt so that we can reduce the number of branches in the top level host CMake.

Reply:
I like that approach, although I'd suggest that you could do it separately from this PR, as it's a bit of technical debt cleanup.

Reply:
I simplified the cmake a bit to make it easier to follow, but we should fix it properly later. I'll make an issue for it.

Reply:
Made the issue here: https://github.com/openenclave/openenclave/issues/2108